.github/workflows/build-macos.yml

```diff
@@ -0,0 +1,133 @@
+name: Build and test on macos
+
+on:
+  pull_request:
```

Comment:
Do we want to trigger a macOS build every time we create a pull request and push on it? How long does it take now to complete this workflow?